students/HABTAMU/lesson06/assignment/note_good_per.txt

```diff
@@ -0,0 +1,37 @@
+Here is the evidence from the cProfile output below, as each module's components call each took the total time spent in and the cumulative time spent in.
+
+Tried to eliminate redundant loops, unnecessary file opening and append function which is the main one is call out 1000012 times.
+And, from 1087687 function calls (1087670 primitive calls) to 44641 function calls (44624 primitive calls) almost three times and even time wize from 8.071 to 3.327 seconds.
+
+
+$ python3 -m cProfile good_perf.py
+{'2015': 5994, '2013': 5911, '2014': 5854, '2016': 5762, '2017': 5789, '2018': 5811}
+'ao' was found 63395 times
+         44641 function calls (44624 primitive calls) in 3.327 seconds
+
+   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
+    21525    0.013    0.000    0.034    0.000 codecs.py:319(decode)
+        6    0.000    0.000    0.000    0.000 good_perf.py:11(<lambda>)
+        1    0.000    0.000    3.335    3.335 good_perf.py:3(<module>)
+        1    0.000    0.000    3.332    3.332 good_perf.py:31(main)
+        1    3.298    3.298    3.332    3.332 good_perf.py:9(analyze)
+    21525    0.021    0.000    0.021    0.000 {built-in method _codecs.utf_8_decode}
+
+       12    0.000    0.000    0.000    0.000 {method 'append' of 'list' objects}
```

Comment:
Looks like you improved things a lot. :) I'm confused, though. Where is the actual code that produced this?

Reply:
Oh, I already gave you credit for assignment 6, so I probably already looked at it, and it didn't change. Nevermind.